src/main/java/BadApple/task/Parser.java

```diff
@@ -0,0 +1,108 @@
+package BadApple.task;
+
+import BadApple.main.BadAppleException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static java.lang.Integer.parseInt;
+
+public class Parser {
+    /**
+     * Takes in a user query and performs the necessary reading and writing to a file on the drive.
+     * The file should exist, and is by default handled by BadPingGuo
+     * @param s the query the user would like to make
+     * @param file the file to read or write from
+     */
+    public static void ProcessQuery(String s, File file) throws IOException {
```

Comment:
Any reason for not using camelCase for `ProcessQuery(...)` method? Would it be better to change this to `processQuery(...)`?